source/net/Address.ooc

```diff
@@ -350,39 +331,34 @@ SocketAddress: abstract class {
 	family: abstract func -> Int
 	host: abstract func -> IPAddress
 	port: abstract func -> Int
-
 	addr: abstract func -> SockAddr*
 	length: abstract func -> UInt
-
 	toString: func -> String {
 		"[%s]:%d" format(host() toString() toCString(), port())
 	}
 	new: static func (host: IPAddress, nPort: Int) -> This {
+		result: This
 		if (host family == AddressFamily IP4) {
 			ip4Host := host as IP4Address
-			return SocketAddressIP4 new(ip4Host ai, nPort)
+			result = SocketAddressIP4 new(ip4Host ai, nPort)
 		}
 		else if (host family == AddressFamily IP6) {
```

Comment:
The `else if` could be moved to the line above. The same applies for the else below.